### ISSUE TITLE
ReferenceError Error: After launch Moonshine in Getting Started page

### ISSUE THREAD

**Issue by piotrzarzycki21** [Moonshine-IDE/Moonshine-IDE#1112]
After launching Moonshine "Getting Started" page shows up. Once we scroll down content we are getting following exception.

```
: Error #1069
: ReferenceError: Error #1069
: 	at moonshine.components.renderers::PackageDependencyRenderer/updateFields()
: 	at moonshine.components.renderers::PackageDependencyRenderer/update()
: 	at feathers.core::ValidatingSprite/validateNow()
: 	at feathers.core::FeathersControl/validateNow()
: 	at feathers.controls::LayoutGroup/validateNow()
: 	at feathers.layout::VerticalListLayout/calculateMaxItemWidth()
: 	at feathers.layout::VerticalListLayout/layout()
: 	at feathers.controls.supportClasses::AdvancedLayoutViewPort/update()
: 	at feathers.core::ValidatingSprite/validateNow()
: 	at feathers.core::FeathersControl/validateNow()
: 	at feathers.controls.supportClasses::BaseScrollContainer/refreshViewPortBoundsForMeasurement()
: 	at feathers.controls.supportClasses::BaseScrollContainer/refreshViewPort()
: 	at feathers.controls.supportClasses::BaseScrollContainer/update()
: 	at feathers.controls::ListView/update()
: 	at feathers.core::ValidatingSprite/validateNow()
: 	at feathers.core::FeathersControl/validateNow()
: 	at feathers.layout::HorizontalLayout/layout()
: 	at feathers.controls::LayoutGroup/handleCustomLayout()
: 	at feathers.controls::LayoutGroup/update()
: 	at moonshine.components.renderers::PackageRenderer/update()
: 	at feathers.core::ValidatingSprite/validateNow()
: 	at feathers.core::FeathersControl/validateNow()
: 	at feathers.controls::LayoutGroup/validateNow()
: 	at feathers.layout::VerticalListLayout/calculateMaxItemWidth()
: 	at feathers.layout::VerticalListLayout/layout()
: 	at feathers.controls.supportClasses::AdvancedLayoutViewPort/update()
: 	at feathers.core::ValidatingSprite/validateNow()
: 	at feathers.core::FeathersControl/validateNow()
: 	at feathers.controls.supportClasses::BaseScrollContainer/refreshViewPortBoundsForMeasurement()
: 	at feathers.controls.supportClasses::BaseScrollContainer/refreshViewPort()
: 	at feathers.controls.supportClasses::BaseScrollContainer/update()
: 	at feathers.controls::ListView/update()
: 	at feathers.core::ValidatingSprite/validateNow()
: 	at feathers.core::FeathersControl/validateNow()
: 	at feathers.core::ValidationQueue/validateNow()
: 	at feathers.core::ValidationQueue/validationQueue_stage_renderHandler()
```

Additional video which shows issue:

https://user-images.githubusercontent.com/24554795/198231730-c2968594-faa4-47b0-bdd4-80591702831e.mov

**Comment by Igazine** [comment]
I'm not able to reproduce this error, although the function where it occurs is quite simple, so I can take a guess what should be changed, but it's difficult to test the results.

@rat-moonshine could you please take a look at it and let me know if this happens to you? Thank you

**Comment by rat-moonshine** [comment]
`PackageDependencyRenderer` is part of MoonshineSDKInstallerGUICore package. I shall give this a try shortly.

**Comment by rat-moonshine** [comment]
On a short update, I didn't reached to the point to receive the error @piotrzarzycki21 reported. But I'm receiving different errors while trying the interface through Getting Started, like,

`ReferenceError: Error #1069: Property get_isAlreadyDownloaded not found on actionScripts.valueObjects.ComponentVO and there is no default value.`

I confirm I don't receive any error when running the interface in MSDKI, though. So I presume this because the more parts of Moonshine is now transferred into Haxe. The pointed get/set methods are also properly declared in [ComponentVO.as](https://github.com/Moonshine-IDE/Moonshine-SDK-Installer/blob/master/InstallerSharedCore/src/actionScripts/valueObjects/ComponentVO.as), as well. (`ComponentVO.as` access by Haxe codes through externs)

This error I have seen in past during GBAuth/Native migration. I remember I started a discussion with @joshtynjala on this. The discussion was like 'what cooks between generating native getter/setters and the need of get_ set_function while declaring `@:flash.property` meta'. 

Cut things short, we found a patch fix (for now) as there were no solid solution came-up by the time, and that fix the problem in GBAuth/Native. 

Before I'm going to have those fix applied in MSDKI codes, I'd ask @Igazine do you have any idea in your mind that could resolve this problem, maybe by altering compilation ways etc. and have this fixed (?) Since I'm not seeing this trouble (yet) in MSDKI itself, so I assume the problem exclusively introduced in Moonshine with the way of its code-usage/compilation or by something.

**Comment by rat-moonshine** [comment]
Or maybe, on a second thought, I should transfer a few classes to Haxe completely now.

**Comment by rat-moonshine** [comment]
> Or maybe, on a second thought, I should transfer a few classes into Haxe completely now.

I have created a sub-issue to track this https://github.com/Moonshine-IDE/Moonshine-SDK-Installer/issues/96.

**Comment by rat-moonshine** [comment]
@Igazine why we have different MSDKI classes converted into [MoonshineGUICore](https://github.com/Moonshine-IDE/Moonshine-IDE/tree/master/ide/MoonshineGUICore/src/actionScripts/valueObjects) (?) 

I also would like to know why you required to have them converted when their corresponding AS3 implementation already in-use in MSDKI source.

**Comment by Igazine** [comment]
It would be really great if Moonshine were programmed in a modern fashion where the entire code-base (classes, interfaces, assets etc.) followed at least a portion of basic OOP-concepts or waterfall-model.

But since there are different projects within the same repo, each of them with their own build configurations, code-base, logic, environment requirements etc., it's really difficult just to even comprehend what's going on in the code, and most importantly why this silly logic is in place. Project1 references Project2's code, Project2 references Project3's code, Project3 references Project1 and this goes on and on. It would be great to have common code-base, projects clearly separated from each other etc., but it's not gonna happen, so I have to make the best out of the current situation.

I understand that it was built this way, all I can do is to follow the current logic and make sure everything builds fine (which it does locally since quite a long time now, I would not commit something that doesn't even build).

Because of all above the mentioned, this transition period (when we're still targeting Flash/Flex with Haxe code) will be very difficult. It's clear that the merge was premature, and now all these errors and issues are present because nobody wanted to test different builds (question is why we have different builds then?)

So yeah, you might run into weird things, please create separate issues for each other, then I'll try to work on them the best I could.

**Comment by Igazine** [comment]
MSDKI (assuming that it's MoonshineSDKInstaller) also has a [separate branch for Haxe conversion](https://github.com/Moonshine-IDE/Moonshine-SDK-Installer/tree/features/haxeconversion), maybe that's why you're experiencing the previously mentioned issue, because you're still on master

**Comment by piotrzarzycki21** [comment]
> Because of all above the mentioned, this transition period (when we're still targeting Flash/Flex with Haxe code) will be very difficult. It's clear that the merge was premature,

I strongly disagree with premature statement. Now after the merge you have more people who try Moonshine on daily bases, you may have even real users to test if someone is using dev version. Thanks to that we have more issues discovered and can react appropriately. Additionally from developer point of view - we are all need to learn Haxe and there is no way that I would learn it by jumping on a branch and maybe do something. Now I'm more forced to do because it's part of my daily development which is another positive point of that merge.

**Comment by rat-moonshine** [comment]
> So yeah, you might run into weird things, please create separate issues for each other, then I'll try to work on them the best I can

It is entirely alright if we see weird things, post-conversion/merge. We are all here to fine-tune/fix/improve them only.

I also respect your opinion on code practices/structure etc. I also honestly, sometime when I looked into these after so many years, so many projects/libraries - eerk me. By that time the project goes through so many implementation, changes that we even thought about (at that time), re-structure multiple times based on a particular time's need, and through so many different people with their respective knowledge-base, we simply can't look into the things now and think it could have design more wisely. Because we never thought about stuffs could be here those are here now.

Again, this is only a sharing, and not justifying against your opinion or frustration (if any).

**Comment by rat-moonshine** [comment]
Some VOs in MSDKI now moved into Haxe. This seems to be fixed the problem reported by @piotrzarzycki21 . 

Requisite changes applied in both Moonshine and MSDKI. I didn't noticed any error during accessing the interfaces from either side. Please give a test with newer Moonshine builds. Let me know if you see any further error in Getting Started or in About.

**Comment by piotrzarzycki21** [comment]
Doesn't occur anymore.
